### PR TITLE
Update default container security context

### PR DESCRIPTION
**Pull request by ChenYi015**
## Purpose of this PR

**Proposed changes:**
- Update default container security context

## Change Category
Indicate the type of change by marking the applicable boxes:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

## Checklist
Before submitting your PR, please review the following:

- [x] I have conducted a self-review of my own code.
- [ ] I have updated documentation accordingly.
- [ ] I have added tests that prove my changes are effective or that my feature works.
- [ ] Existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by jacobsalway** [comment]
Might need to do some investigation to map out all the disk usage of `spark-submit` but what are your thoughts on also using an `emptyDir` volume in the controller and adding `readOnlyRootFilesystem: true` as well?

File: charts/spark-operator-chart/values.yaml [hunk 1]
```diff
@@ -156,10 +156,14 @@ controller:
...
 
   # -- Security context for controller containers.
-  securityContext: {}
-    # runAsUser: 1000
-    # runAsGroup: 2000
-    # fsGroup: 3000
+  securityContext:
+    runAsNonRoot: true
+    runAsUser: 185
+    runAsGroup: 185
```
Comment (ImpSy) [inline charts/spark-operator-chart/values.yaml:162]: Hardcoding UserID and GroupID feels a bit to static
If we ever decide to change our base image those could change

But requiring `runAsNonRoot`, setting an FsGroup and dropping all capabilities is a great idea

**Comment by ImpSy** [comment]
> Might need to do some investigation to map out all the disk usage of spark-submit but what are your thoughts on also using an emptyDir volume in the controller and adding readOnlyRootFilesystem: true as well?

Yeah I feel like `readOnlyRootFilesystem` is a bit too restrictive especially while we are actively modifying the usage of the controller with pod templates

But like the PR #2219 propose we could mount a  volume on /tmp and use `os.CreateTemp` everytime we need a local file

**Comment by jacobsalway** [comment]
Could push the user and group directives down to the Dockerfile rather than the container security context. I'm not sold on read-only FS because any temporary disk usage could cause a panic but it's an enhancement we can make later down the line if requested/it's a common security requirement or implemented in the other PR.

File: charts/spark-operator-chart/values.yaml [hunk 2]
```diff
@@ -156,10 +156,14 @@ controller:
     #   memory: 300Mi
 
   # -- Security context for controller containers.
-  securityContext: {}
-    # runAsUser: 1000
-    # runAsGroup: 2000
-    # fsGroup: 3000
+  securityContext:
+    runAsNonRoot: true
```
Comment (jacobsalway) [inline charts/spark-operator-chart/values.yaml:160]: Could add `allowPrivilegeEscalation: false` as well

Comment (ChenYi015) [reply charts/spark-operator-chart/values.yaml:160]: I did not add this as the default value of `allowPrivilegeEscalation` should be `false`.

Comment (ChenYi015) [reply charts/spark-operator-chart/values.yaml:160]: Done.

**Comment by ChenYi015** [comment]
> Could push the user and group directives down to the Dockerfile rather than the container security context. I'm not sold on read-only FS because any temporary disk usage that isn't on a volume could cause a panic but it's an enhancement we can make later down the line if requested/it's a common security requirement or implemented in the other PR.

I have pushed the user and group directives down  to the Dockerfile. Agree on that we would better implement the readOnlyRootFileSystem in a dedicated PR.

File: charts/spark-operator-chart/values.yaml (hunk 1 above)
Comment (ChenYi015) [reply charts/spark-operator-chart/values.yaml]: Have removed the `runAsUser` and `runAsGroup` and pushed them down to the Dockerfile.

**Review by jacobsalway — APPROVED** [review]
/lgtm

Works for me in local testing on Kind. As you said let's do `readOnlyRootFilesystem: true` in a separate PR.

**Comment by cccsss01** [comment]
> > Might need to do some investigation to map out all the disk usage of spark-submit but what are your thoughts on also using an emptyDir volume in the controller and adding readOnlyRootFilesystem: true as well?
> 
> Yeah I feel like `readOnlyRootFilesystem` is a bit too restrictive especially while we are actively modifying the usage of the controller with pod templates
> 
> But like the PR #2219 propose we could mount a volume on /tmp and use `os.CreateTemp` everytime we need a local file

majority of these local files created don't need to be stored permanently correct?

**Pull request merged**